src/core/annotation.js

```diff
@@ -1081,6 +1094,92 @@ class WidgetAnnotation extends Annotation {
 
     return `${shift} ${vPadding} Td (${text}) Tj`;
   }
+
+  handleMultiline(text, font, fontSize, width, alignment, hPadding, vPadding) {
+    const lines = text.replace("\r\n", "\n").split("\n");
+    let buf = "";
+    const totalWidth = alignment === 1 ? width : width - 2 * hPadding;
+    for (const line of lines) {
+      const chunks = this.splitLine(line, font, fontSize, totalWidth);
+      for (const chunk of chunks) {
+        if (buf === "") {
+          buf = this.renderPDFText(
+            chunk,
+            font,
+            fontSize,
+            width,
+            alignment,
+            hPadding,
+            -fontSize
+          );
+        } else {
+          buf +=
+            "\n" +
+            this.renderPDFText(
+              chunk,
+              font,
+              fontSize,
+              width,
+              alignment,
+              0,
+              -fontSize
+            );
+        }
+      }
+    }
+
+    return buf;
+  }
+
+  splitLine(line, font, fontSize, width) {
+    const scale = fontSize / 1000;
+    const white = font.charsToGlyphs(" ", true)[0].width * scale;
+    const chunks = [];
+
+    let lastSpacePos = -1,
+      startChunk = 0,
+      currentWidth = 0;
+
+    for (let i = 0; i < line.length; i++) {
```

Comment:
We usually do `for (let i = 0, ii = line.length; i < ii; i++) {` to cache the line length once.